### PR TITLE
MCOL-4188 Regression fixes for MCOL-641.

### PR DESCRIPTION
**Pull request by tntnatbry**
1. Make PredicateOperator::setOpType() function wide decimal aware.
  2. Added support for wide decimal in jlf_subquery.cpp::getColumnValue()
     used in scalar subqueries.
  3. Fixed the column index used for fetching wide decimal values from a Row
     when a wide decimal field is used in the order by clause.

### CODE REVIEW

File: dbcon/joblist/jlf_subquery.cpp
Comment (tntnatbry) [inline dbcon/joblist/jlf_subquery.cpp]: I will remove the `row.getInt128Field()` in favour of `row.getTSInt128Field()` once the new IDB_Decimal ctor taking a `TSInt128` as an argument is available as part of https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/1610

**Comment by tntnatbry** [comment]
This PR needs an update which is dependent on the merge of https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/1610.

File: dbcon/execplan/predicateoperator.cpp
Comment (drrtuy) [inline dbcon/execplan/predicateoperator.cpp]: Why not use std::max() here?

File: dbcon/joblist/jlf_subquery.cpp
Comment (drrtuy) [inline dbcon/joblist/jlf_subquery.cpp]: I believe we don't need oss here.

File: dbcon/joblist/jlf_subquery.cpp
Comment (drrtuy) [reply dbcon/joblist/jlf_subquery.cpp]: You can even use a ctor with 3 arguments that doesn't set the int64 value in the Decimal

File: dbcon/joblist/jlf_subquery.cpp
Comment (drrtuy) [inline dbcon/joblist/jlf_subquery.cpp]: Ohh, I didn't realize what you are doing here.
I've moved dataconvert::decimalToString into Decimal::toString() so it can print out both narrow and wide decimals and you need to only call Decimal ctor with appropriate arguments and call Decimal::toString() of the resulting instance calling ConstantColumn ctor.

**Comment by drrtuy** [comment]
Good job. 
I don't see other issues with the PR so we can proceed after the comments will be addressed.

File: dbcon/execplan/predicateoperator.cpp
Comment (tntnatbry) [reply dbcon/execplan/predicateoperator.cpp]: Good point. Done.

File: dbcon/joblist/jlf_subquery.cpp
Comment (tntnatbry) [reply dbcon/joblist/jlf_subquery.cpp]: Done.

File: dbcon/joblist/jlf_subquery.cpp
Comment (tntnatbry) [reply dbcon/joblist/jlf_subquery.cpp]: I agree. I am now constructing the IDB_Decimal object and calling the narrow decimal version of dec.toString().

**Comment by tntnatbry** [comment]
Comments are addressed. There is one change related to the use of `row.getInt128Field(i, val);` in this patch. This will be addressed once either PR https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/1618 or PR https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/1610 is merged. Until then, please don't merge this PR.

**Review by drrtuy — APPROVED** [review]

**Pull request merged**
